contracts/SpeedMarkets/MultiCollateralOnOffRamp.sol

```diff
@@ -188,6 +223,31 @@ contract MultiCollateralOnOffRamp is Initializable, ProxyOwned, ProxyPausable, P
         require(amountOut <= getMaximumReceived(tokenIn, amountIn), "Amount above max allowed peg slippage");
     }
 
+    function _swapExactInputOfframp(uint256 amountIn, address tokenOut) internal returns (uint256 amountOut) {
+        sUSD.approve(address(swapRouter), amountIn);
+
+        bytes memory pathToUse = pathPerCollateralOfframp[tokenOut];
+        if (pathToUse.length == 0) {
+            uint24 fee = 3000;
+            pathToUse = abi.encodePacked(address(sUSD), fee, WETH9, fee, address(tokenOut));
+        }
+
+        // Multiple pool swaps are encoded through bytes called a `path`. A path is a sequence of token addresses and poolFees that define the pools used in the swaps.
+        ISwapRouter.ExactInputParams memory params = ISwapRouter.ExactInputParams({
+            path: pathToUse,
+            recipient: address(this),
+            deadline: block.timestamp + 15,
+            amountIn: amountIn,
+            amountOutMinimum: getMinimumReceivedOfframp(tokenOut, amountIn)
+        });
+
+        // The call to `exactInput` executes the swap.
+        amountOut = swapRouter.exactInput(params);
```

Comment:
possible gas savings if the `ISwapRouter.ExactInputParams` struct is created directly in the `exactInput()` call
```
amountOut = swapRouter.exactInput(ISwapRouter.ExactInputParams({
 path: pathToUse, 
 ...
 }));
```

Reply:
avoids storing in memory than copying to the external call, although that might be exactly the same in anyways. 
So no hard opinion on this.